File: java/src/main/java/com/mars/vpp/client/sample/java/VppClient.java
package com.mars.vpp.client.sample.java;

import io.fd.vpp.jvpp.JVppRegistryImpl;
import io.fd.vpp.jvpp.core.JVppCore;
import io.fd.vpp.jvpp.core.JVppCoreImpl;
import io.fd.vpp.jvpp.core.dto.BridgeDomainAddDel;
import io.fd.vpp.jvpp.core.dto.BridgeDomainAddDelReply;
import io.fd.vpp.jvpp.core.dto.BridgeDomainDump;
import io.fd.vpp.jvpp.core.dto.CreateLoopback;
import io.fd.vpp.jvpp.core.dto.CreateLoopbackReply;
import io.fd.vpp.jvpp.core.dto.SwInterfaceDump;
import io.fd.vpp.jvpp.core.future.FutureJVppCoreFacade;

public class VppClient {

    public static final String CLIENT_ID = "Vppclient";

    public static void main(String[] args) throws Exception {

        try (JVppRegistryImpl registry = new JVppRegistryImpl(CLIENT_ID);

             JVppCore jVppCore = new JVppCoreImpl();
             FutureJVppCoreFacade coreFacade = new FutureJVppCoreFacade(registry, jVppCore)) {

            // Dump interfaces
            dumpInterfaces(coreFacade);

            // Create loopback
            createLoopback(coreFacade);

            // Dump interfaces
            dumpInterfaces(coreFacade);

            // Dump bds
            dumpBds(coreFacade);

            // Create bd
            createBd(coreFacade);

            // Dump bds
            dumpBds(coreFacade);
        }
    }

    private static void createLoopback(FutureJVppCoreFacade coreFacade) throws Exception {
        CreateLoopback createLoopback = new CreateLoopback();
        System.out.printf("Sending create loopback. Msg id: %s\n", createLoopback.getClass().getSimpleName());
        CreateLoopbackReply createLoopbackReply = coreFacade.createLoopback(createLoopback)
                .toCompletableFuture()
                .get();

        System.out.printf("\tCreate loopback reply received, message id: %s, new interface index: %d\n",
                createLoopbackReply.getClass().getSimpleName(), createLoopbackReply.swIfIndex);
    }

    private static void createBd(FutureJVppCoreFacade coreFacade) throws Exception {
        BridgeDomainAddDel createBd = new BridgeDomainAddDel();
        createBd.bdId = 99;
        createBd.isAdd = 1;
        System.out.printf("Sending create bd. Msg id: %s\n", createBd.getClass().getSimpleName());
        BridgeDomainAddDelReply createBdReply = coreFacade.bridgeDomainAddDel(createBd)
                .toCompletableFuture()
                .get();

        System.out.printf("\tCreate bd reply received, message id: %s\n",
                createBdReply.getClass().getSimpleName());
    }

    private static void dumpInterfaces(FutureJVppCoreFacade coreFacade) throws Exception {
        SwInterfaceDump swInterfaceDump = new SwInterfaceDump();
        System.out.printf("Sending dump interfaces. Msg id: %s\n", swInterfaceDump.getClass().getSimpleName());
        coreFacade.swInterfaceDump(swInterfaceDump)
                .toCompletableFuture()
                .get()
                .swInterfaceDetails
                .forEach(ifcDetail -> System.out.printf(
                        "\tInterface, message id: %s, interface index: %s\n",
                        ifcDetail.getClass().getSimpleName(), ifcDetail.swIfIndex));
    }

    private static void dumpBds(FutureJVppCoreFacade coreFacade) throws Exception {
        BridgeDomainDump bdDump = new BridgeDomainDump();
        bdDump.bdId = ~0;
        System.out.printf("Sending dump bridge domains. Msg id: %s\n", bdDump.getClass().getSimpleName());
        coreFacade.bridgeDomainDump(bdDump)
                .toCompletableFuture()
                .get()
                .bridgeDomainDetails
                .forEach(ifcDetail -> System.out.printf(
                        "\tBridge domain, message id: %s, bd index: %s\n",
                        ifcDetail.getClass().getSimpleName(), ifcDetail.bdId));
    }
}

File: kotlin/src/main/kotlin/com/mars/vpp/client/sample/kotlin/VppClient.kt
package com.mars.vpp.client.sample.kotlin

import io.fd.vpp.jvpp.JVppRegistryImpl
import io.fd.vpp.jvpp.core.JVppCoreImpl
import io.fd.vpp.jvpp.core.dto.BridgeDomainAddDel
import io.fd.vpp.jvpp.core.dto.BridgeDomainDump
import io.fd.vpp.jvpp.core.dto.CreateLoopback
import io.fd.vpp.jvpp.core.dto.SwInterfaceDump
import io.fd.vpp.jvpp.core.future.FutureJVppCoreFacade

object VppClient {

    private const val CLIENT_ID = "Vppclient"

    @JvmStatic
    fun main(args: Array<String>) {

        JVppRegistryImpl(CLIENT_ID).use { registry ->
            JVppCoreImpl().use { jVppCore ->
                FutureJVppCoreFacade(registry, jVppCore).use { coreFacade ->

                    // Dump interfaces
                    dumpInterfaces(coreFacade)

                    // Create loopback
                    createLoopback(coreFacade)

                    // Dump interfaces
                    dumpInterfaces(coreFacade)

                    // Dump bds
                    dumpBds(coreFacade)

                    // Create bd
                    createBd(coreFacade)

                    // Dump bds
                    dumpBds(coreFacade)
                }
            }
        }
    }

    private fun createLoopback(coreFacade: FutureJVppCoreFacade) {
        CreateLoopback().let { request ->
            println("Sending create loopback. Msg id: ${request.javaClass.simpleName}")
            coreFacade.createLoopback(request)
                    .toCompletableFuture()
                    .get()
                    .let {
                        println("\tCreate loopback reply received, message id: " +
                                "${it.javaClass.simpleName}, new interface index: ${it.swIfIndex}")
                    }
        }
    }

    private fun createBd(coreFacade: FutureJVppCoreFacade) {
        BridgeDomainAddDel().let { request ->
            request.bdId = 99
            request.isAdd = 1
            println("Sending create bd. Msg id: ${request.javaClass.simpleName}")
            coreFacade.bridgeDomainAddDel(request)
                    .toCompletableFuture()
                    .get()
                    .let {
                        println("\tCreate bd reply received, message id: ${it.javaClass.simpleName}")
                    }
        }
    }

    private fun dumpInterfaces(coreFacade: FutureJVppCoreFacade) {
        SwInterfaceDump().let { request ->
            println("Sending dump interfaces. Msg id: ${request.javaClass.simpleName}")
            coreFacade.swInterfaceDump(request)
                    .toCompletableFuture()
                    .get()
                    .swInterfaceDetails
                    .forEach {
                        println("\tInterface, message id: ${it.javaClass.simpleName}, interface index: ${it.swIfIndex}")
                    }
        }
    }

    private fun dumpBds(coreFacade: FutureJVppCoreFacade) {
        BridgeDomainDump().let { request ->
            request.bdId = 0.inv()
            println("Sending dump bridge domains. Msg id: ${request.javaClass.simpleName}")
            coreFacade.bridgeDomainDump(request)
                    .toCompletableFuture()
                    .get()
                    .bridgeDomainDetails
                    .forEach {
                        println("\tBridge domain, message id: ${it.javaClass.simpleName}, bd index: ${it.bdId}")
                    }
        }
    }
}
